swarm/api/http/error.go

```diff
@@ -29,12 +29,19 @@ import (
 	"time"
 
 	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/metrics"
 	"github.com/ethereum/go-ethereum/swarm/api"
 )
 
 //templateMap holds a mapping of an HTTP error code to a template
 var templateMap map[int]*template.Template
 
+//metrics variables
+var (
+	htmlCounter = metrics.NewCounter("api.http.errorpage.html.count")
```

Comment:
Do we need to be this granular? Should we not just log the count of 5xx errors?

Reply:
Thinking of it, it makes most sense to me to actually track how many requests were being made for JSON and how many for HTML. This is currently missing as I couldn't identify a specific spot where this would be easy. For the time being this could be kept, as I anyway expect 99.9% to be HTML (but that's why this could be interesting to measure, as I don't know).

Reply:
If you want to count the amount of HTML responses, check `http.ResponseWriter.Header` for the `Content-Type` containing `text/html` just before returning a response in `ServeHTTP`.

Reply:
The latter seems actually to not be very practical, as the current code for `ServeHTTP` returns at many different places.